routes/file.js

```diff
@@ -3,89 +3,102 @@
 var cvr = require( "cvr" );
 var models = require( "../lib/models" );
 var env = require( "../lib/env" );
+var a = require( "async" );
+
+var createError = function ( message, status )
+{
+    var err = new Error( message );
+    err.status = status;
+    return err;
+};
 
 module.exports = function ( req, res, next )
 {
-    var onRepo = function ( err, repo )
-    {
-        if( err )
-        {
-            return next( err );
-        }
+    var repo = null;
+    var commit = null;
+    var cov = null;
+    var fileCov = null;
 
-        if( !repo )
+    a.waterfall( [
+        function ( done )
         {
-            return res.status( 404 );
-        }
-
-        var onCommit = function ( err, commit )
+            models.Repo.findByOwnerAndName( req.params.owner, req.params.name, done );
+        },
+        function ( _repo, done )
         {
-            if( err )
+            repo = _repo;
+
+            if( !repo )
             {
-                return next( err );
+                throw createError( "Repo not found", 404 );
             }
 
+            models.Commit.findCommit( repo.owner, repo.name, req.params.hash, done );
+        },
+        function ( _commit, done )
+        {
+            commit = _commit;
+
             if( !commit )
             {
-                return res.status( 404 ).end();
+                throw createError( "Commit not found", 404 );
```

Comment:
Why `throw` instead of passing the error object to the waterfall task's `done` callback? Then handle the error in the waterfall final callback.